client/src/modules/MentorRegistry/components/MentorRegistryTableContainer.tsx

```diff
@@ -0,0 +1,349 @@
+    return (values: number[], record: MentorRegistryDto) => {
+      return (
+        <>
+          {values
+            .map(v => ({
+              value: courses.find(c => c.id === v)?.name ?? v.toString(),
+              alias: courses.find(c => c.id === v)?.alias ?? '',
+              color: record.courses.includes(v) ? '#87d068' : undefined,
+            }))
+            .map(v => (v.color ? colorTagRenderer(v.value, v.color) : renderTagWithCopyButton(v.value, v.alias)))}
+        </>
+      );
+    };
+  };
+
+  const renderTagWithCopyButton = (value: string, alias: string) => {
+    const link = `${window.location.origin}/course/mentor/confirm?course=${alias}`;
+    return (
+      <>
+        {colorTagRenderer(value)} <CopyToClipboardButton value={link} type="link" />
+      </>
+    );
+  };
+
+  const renderInfo = (_: any, record: MentorRegistryDto) => {
+    const isMentor = record.courses.some(id => !record.preselectedCourses.includes(id));
+    return (
+      <div className="info-icons">
+        {isMentor ? <div title="Mentor in the past" className="icon-mentor" /> : null}
+        {record.hasCertificate ? (
+          <SafetyCertificateTwoTone
+            title="Completed with certificate"
+            className="icon-certificate"
+            twoToneColor="#52c41a"
+          />
+        ) : null}
+        <style jsx>{mentorRegistryStyles}</style>
+      </div>
+    );
+  };
+
+  const filteredData = useMemo(() => {
+    return mentors.filter(
+      mentor =>
+        (combinedFilter.technicalMentoring.length
+          ? mentor.technicalMentoring.some(value => combinedFilter.technicalMentoring.includes(value))
+          : true) &&
+        (combinedFilter.preselectedCourses.length
+          ? mentor.preselectedCourses.some(value => combinedFilter.preselectedCourses.includes(value))
+          : true) &&
+        (combinedFilter.preferredCourses.length
+          ? mentor.preferedCourses.some(value => combinedFilter.preferredCourses.includes(value))
+          : true),
+    );
+  }, [combinedFilter, mentors]);
+
+  const handleTableChange = (_: any, filters: Record<MentorsRegistryColumnKey, FilterValue | string[] | null>) => {
+    const combinedFilter: CombinedFilter = {
+      preferredCourses: filters.preferedCourses?.map(c => Number(c)) ?? [],
+      preselectedCourses: filters.preselectedCourses?.map(c => Number(c)) ?? [],
+      technicalMentoring: filters.technicalMentoring?.map(v => v.toString()) ?? [],
+    };
+
+    const filterTag: string[] = [
+      ...combinedFilter.technicalMentoring.map(v => `${MentorsRegistryColumnName.Tech}: ${v}`),
+      ...combinedFilter.preselectedCourses.map(
+        v => `${MentorsRegistryColumnName.Preselected}: ${courses.find(i => i.id === v)?.name}`,
+      ),
+      ...combinedFilter.preferredCourses.map(
+        v => `${MentorsRegistryColumnName.PreferredCourses}: ${courses.find(i => i.id === v)?.name}`,
+      ),
+    ];
```

Comment:
Please consider to use more meaningful names instead of `c`, `v`, `i` etc. here and in other places in scope of this PR

Reply:
fixed